feat: recover payments on failed days

Tiavina-Andriamamivony opened the pull request.

Tiavina-Andriamamivony committed:
feat(recovery): add service and endpoint for Orange transaction recovery

Added OrangeTransactionRecoveryService to fetch Orange transactions for a given date, insert missing transactions, and trigger corresponding payment verifications via PaymentVerificationRequestedService. Added OrangeRecoveryController with POST /api/orange/recover?date=YYYY-MM-DD endpoint to trigger recovery and return a JSON summary. Enhanced logging for fetch, insert, and payment verification steps. Added unit tests: OrangeTransactionRecoveryServiceTest (Mockito) for service logic and OrangeRecoveryControllerTest (MockMvc) for endpoint behavior. Notes: endpoint is idempotent and should be secured before exposure.

Tiavina-Andriamamivony committed:
feat: find payments feature

Tiavina-Andriamamivony committed:
chore: format

Tiavina-Andriamamivony committed:
chore: replace env conf data

Tiavina-Andriamamivony committed:
chore(test): replace test data

Tiavina-Andriamamivony committed:
chore: format

Tiavina-Andriamamivony committed:
Merge branch 'hei-school:preprod' into preprod

Tiavina-Andriamamivony committed:
Merge branch 'verifyAll' into preprod

Tiavina-Andriamamivony committed:
chore: delete comments

Tiavina-Andriamamivony committed:
chore: clean

Tiavina-Andriamamivony committed:
fix: multiple payment verification

Delete custom jrepository method, return failed payments

Tiavina-Andriamamivony committed:
chore: format

Tiavina-Andriamamivony committed:
test: expect FAILED payments for non-existing payment infos

hei-teacher commented on src/main/java/school/hei/vola/endpoint/rest/controller/OrangeRecoveryController.java:
```diff
@@ -0,0 +1,35 @@
+   * missing orange_transaction rows, - trigger verification for matching payments, - return a JSON
+   * summary per transaction.
+   */
+  @PostMapping("/recover")
```
Post... Should be idempotent with Put.

recover... is not rest

hei-teacher commented on src/main/java/school/hei/vola/endpoint/rest/controller/OrangeRecoveryController.java:
```diff
@@ -0,0 +1,35 @@
+   * summary per transaction.
+   */
+  @PostMapping("/recover")
+  public ResponseEntity<List<OrangeTransactionRecoveryService.RecoveryResult>> recover(
```
ResponseEntity seems superfluous

hei-teacher commented on src/main/java/school/hei/vola/repository/PaymentRepository.java:
```diff
@@ -75,4 +81,54 @@ public Optional<Payment> findPaymentByPayerEmailAndPspTypeAndPspPaymentId(
+
+    return paymentInfos.stream()
+        .map(
+            info -> {
```
Bad style...

hei-teacher commented on src/main/java/school/hei/vola/service/OrangeTransactionRecoveryService.java:
```diff
@@ -0,0 +1,113 @@
+   * orange_transaction - trigger verification for any matching payments - return a list of
+   * RecoveryResult for each transaction processed
+   */
+  public List<RecoveryResult> recover(LocalDate date) {
```
Bad style

Tiavina-Andriamamivony committed:
chore(recovery): set endpoint type to put and replace response entity

Tiavina-Andriamamivony committed:
chore(orange): simplify transaction recovery service and improve code readability

- Remove inner RecoveryResult class in favor of sync.model.RecoveryResult
- Extract methods to follow Single Responsibility Principle
- Add meaningful method names
- Improve error handling with granular try-catch blocks
- Separate concerns: persistence, verification, and result building
- Update tests to use new RecoveryResult model and sync() method
- Add test cases for failure and partial success scenarios

Tiavina-Andriamamivony committed:
chore: format

Tiavina-Andriamamivony committed:
chore: refactor PaymentRepository for better readability

- Extract findPaymentsByPaymentInfos logic into atomic methods (findOrCreateFailedPayment, mapToFoundPayment, createFailedPlaceholderPayment)
- Replace magic number 6 with FAILED_PAYMENT_ATTEMPT_COUNT constant
- Use Optional.map/orElseGet instead of verbose if/else
- Add [PAYMENT_SEARCH] log prefix for consistency
- Add Javadoc to document placeholder FAILED payment behavior
- Improve testability through separation of concerns

Tiavina-Andriamamivony committed:
chore: delete excessive log and add some tests

Tiavina-Andriamamivony committed:
chore: refactor Orange sync service and simplify tests

Tiavina-Andriamamivony committed:
chore: revert randomEmail result

Tiavina-Andriamamivony committed:
chore: refactor Orange sync service and simplify tests

Tiavina-Andriamamivony committed:
test: remove redundant OrangeRecoveryTest

Tiavina-Andriamamivony committed:
chore(orangeSync): refactor
